aion/web/channels/question_chronicle.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Aion.QuestionChronicle do
+  @moduledoc """
+    This module saves the time when questions were
+    changed in specific rooms. Thanks to that when a timer
+    fires, it's easy to check if the question should be changed
+    to a new one.
+  """
+  @type t :: %{binary => integer}
+
+  @question_timeout_seconds 10
+  @question_timeout_micro @question_timeout_seconds * 1_000_000
+
+  def question_timeout_micro, do: @question_timeout_micro
+  def question_timeout_milli, do: div @question_timeout_micro, 1000
+
+  def start_link do
+    Agent.start_link(fn -> %{} end, name: __MODULE__)
+  end
+
+  @doc "Lists all entries stored by the Agent"
+  @spec list_entries :: __MODULE__.t
+  def list_entries, do: Agent.get(__MODULE__, &(&1))
+
+  @doc "Checks if a question should be changed due to a timeout"
+  @spec should_change?(binary) :: boolean
+  def should_change?(room_id), do: should_change?(room_id, &get_current_time/0)
+  @spec should_change?(binary, function) :: boolean
+  def should_change?(room_id, time) do
+    last_changed = Agent.get(__MODULE__, fn chronicle ->
```

Comment:
I see the opportunity to make a function out of it ;)

Reply:
I don't think it is an `opportunity` to create a separate function. You don't need to redefine all the functions that are already defined. Getters and setters may come in handy if the data structure is quite complex. Here, it's a simple map, though and this is what `Map.get` is for. 

The truth is, maybe it shouldn't be written as `fn arg -> arg`, it would look clearer with the & syntax here.

Reply:
last_changed(room_id) looks much cleaner in my opinion, but keep in mind that this is only my point of view ;)

Reply:
Ahh, you meant creating a separate function for reading parts of state from Agent? Yeah, I give you that, since we are calling first Agent and then Map.get, it may actually be a good idea to move it out to a separate function. 

I'm not sure if it's really needed here, though, as long as this is the only place where we want to read Agent's state. I think we should move it out to something like `last_changed(room_id)` as soon as there is another place where we use this functionality.